### PR TITLE
Add syntax highlighting with highlight.js

### PR DESCRIPTION
**Pull request by bmackinney**
Merging this pull request will add syntax highlighting to preformatted code blocks with the [highlight.js](https://highlightjs.org/) library
This is a WIP because language detection seems off. 
This is the output at the configuring settings.php doc, which is accurate:
- [x] Identify all documents with syntax highlighting
- [x] Identify language tagging standards and add to the [style-guide.md](https://github.com/pantheon-systems/documentation/blob/syntax-highlight/style-guide.md)
- [x] Tag languages in code blocks according to style guide

### CODE REVIEW

**Comment by rachelwhitton** [comment]
@bmackinney can I do anything to help out here? I think the best way to proceed is to finalize the custom package with selected languages and merge this. 

I'll volunteer to comb through each doc to ensure the correct language is applied and do a bulk PR for needed changes.

**Comment by rachelwhitton** [comment]
@bmackinney this is ready to merge +1 
Examples:
screen shot 2015-06-09 at 9 40 46 am<hr>
screen shot 2015-06-09 at 9 44 41 am

File: source/docs/articles/drupal/private-files.md [hunk 1]
```diff
@@ -47,13 +47,13 @@ This depends on the workflow and that you are planning to implement. If you set
...
 
 This can be done via [Terminus](https://github.com/pantheon-systems/cli):
-
-    # Set this to Test/Live
-    $: terminus drush --site=<site> --env=<env> vset uc_credit_encryption_path 'private'
-    # verify the path is set on Test/Live
-    $: terminus drush --site=<site> --env=<env> vget uc_credit_encryption_path
-    uc_credit_encryption_path: "private"
-
+```html
```
Comment (bmackinney) [inline source/docs/articles/drupal/private-files.md]: @rachelwhitton why `html` over `bash`?

**Comment by bmackinney** [comment]
@rachelwhitton this is close. only questions I have:
 - Why is so much code marked `nginx` and `html` for what seem to be `shell`? I see some `ruby` in there too.
- Once they're all good to go, we need to resolve the conflicts and deploy.

**Comment by rachelwhitton** [comment]
@bmackinney I picked different languages based on how it displayed the snippet, then I used them throughout for consistency - not ideal but here are some examples where `bash` didn't quite look right:
## Example 1
### `nginx`
screen shot 2015-06-15 at 10 58 54 am
### `bash`
screen shot 2015-06-15 at 10 59 21 am
## Example 2
### `nginx`
screen shot 2015-06-15 at 11 00 02 am
### `bash`
screen shot 2015-06-15 at 10 59 54 am
## Example 3
### `nginx`
screen shot 2015-06-15 at 11 12 48 am
### `bash`
screen shot 2015-06-15 at 11 13 06 am

**Comment by rachelwhitton** [comment]
Todo - 

- [ ] Revise specified languages to use <code>```nohighlight</code> if CSS needs adjusting.

**Comment by bmackinney** [comment]
- [ ] specified languages should be accurate -no making it look better by specifying a different language. Use `bash` or `nohighlight` when the default is incorrectly styled. 
- [ ] fix the conflicts.

**Comment by rachelwhitton** [comment]
@bmackinney ready for review. Updates deployed to the [syntax CDE](http://syntax-panther.pantheon.io/docs)

**Comment by bmackinney** [comment]
Sweet. I'd like to remove the `.hljs block` from [`main.css`](https://github.com/pantheon-systems/documentation/pull/262/files#diff-854e219c2ce2db8dbb6fe7d9d56068f4R1379), because the background color prevents us from using another theme. @ari believes we should use Dark. This is what it looks like with Dark enabled and the four styles disabled. 
image

**Comment by ari-gold** [comment]
Not a blocker, but yeah, some dark theme/style would be nice! =) On https://highlightjs.org/ when clicking on Style it cycles through different themes. Not sure which dark theme is best.

**Comment by rachelwhitton** [comment]
Dark theme deployed to [syntax CDE](http://syntax-panther.pantheon.io/docs/articles/sites/code/redirect-incoming-requests/)
Screenshot:
screen shot 2015-06-17 at 10 55 22 am

@bmackinney the font sizes have changed in the TOC - was that intentional?

**Comment by bmackinney** [comment]
@rachelwhitton I don't see where the font sizes changed. https://github.com/pantheon-systems/documentation/blame/syntax-highlight/source/docs/assets/css/main.css#L711 says the only changes to that part of the template are old/from Nate.

**Comment by rachelwhitton** [comment]
Looks like the line height was changed here: https://github.com/pantheon-systems/documentation/commit/9d6b8b9a742aef3b8e29b00549898b1cccbccf87

That never made it to [master](https://github.com/pantheon-systems/documentation/blob/master/source/docs/assets/css/jquery.tocify.css#L23)

Let me know if any changes should be made @bmackinney 

### Master TOC:
screen shot 2015-06-17 at 12 37 16 pm


### Syntax CDE TOC:
screen shot 2015-06-17 at 12 36 49 pm

**Comment by bmackinney** [comment]
deferring to @informanddelight. this is an old change. consider comparing out https://pantheon.io/docs/articles/getting-started and http://syntax-panther.pantheon.io/docs/articles/getting-started to decide whether to axe the old change to this file.

**Comment by rachelwhitton** [comment]
+1 for keeping the TOC changes.

**Comment by informanddelight** [comment]
+1 for keeping TOC changes too :)
Thanks for asking

**Comment by bmackinney** [comment]
:ship:

**Pull request merged**